packages/smart-contracts/test/contracts/BatchEthereumPayments.test.ts

```diff
@@ -0,0 +1,210 @@
+import { ethers, network } from 'hardhat';
+import { BigNumber, Signer } from 'ethers';
+import { expect, use } from 'chai';
+import { solidity } from 'ethereum-waffle';
+import { EthereumFeeProxy, BatchPayments } from '../../src/types';
+import { batchPaymentsArtifact } from '../../src/lib';
+
+import { ethereumFeeProxyArtifact } from '../../src/lib/';
+import { HttpNetworkConfig } from 'hardhat/types';
+
+use(solidity);
+
+const logInfos = false;
+
+describe('contract: BatchPayments: Ethereum', () => {
```

Comment:
What should the behaviour be if the arrays given to the function are not of the same length? Shouldn't we test that to control the error messages and ease maintenance? (same applies for Batch ERC20 batch tests + contract)

Reply:
added.